Decred Wallet/Custom Views/FloatingLabelInputs/FloatingPlaceholderLabel.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  FloatingPlaceholderLabel.swift
+//  Decred Wallet
+//
+// Copyright (c) 2019 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+import UIKit
+
+@IBDesignable
+class FloatingPlaceholderLabel: UILabel {
+    @IBInspectable var topPadding: CGFloat = 0
+    @IBInspectable var bottomPadding: CGFloat = 0
+    @IBInspectable var leftPadding: CGFloat = 4
+    @IBInspectable var rightPadding: CGFloat = 4
+
+    var topConstraint: NSLayoutConstraint?
+
+    let normalColor: UIColor = UIColor.appColors.lightBluishGray
+    var activeColor: UIColor = UIColor.appColors.lightBlue
+    let normalFontSize: CGFloat = 16
+    let smallerFontSize: CGFloat = 14
+
+    override var text: String? {
+        didSet {
+            // set position after the text did set, since the label position calculated in real-time using the height of the label
+            self.setPosition()
+        }
+    }
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        self.initView()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        self.initView()
+    }
+
+    public func initView() {
+        self.backgroundColor = UIColor.white
+        self.clipsToBounds = true
+        self.translatesAutoresizingMaskIntoConstraints = false
+        self.textColor = self.normalColor
+        self.setPosition()
+    }
+    
+    private func setPosition() {
+        guard let floatingLabelInput = superview as? FloatingLabelProtocol else { return }
+
+        self.font = self.font.withSize(floatingLabelInput.isParentEmpty() ? self.normalFontSize : self.smallerFontSize)
+        self.sizeToFit()
+        let calculatedTopConstraintValue = floatingLabelInput.isParentEmpty() ? (self.superview!.frame.size.height - self.intrinsicContentSize.height) / 2 : self.intrinsicContentSize.height / -2
+
+        if self.topConstraint == nil {
+            self.topConstraint = self.topAnchor.constraint(equalTo: self.superview!.topAnchor, constant: calculatedTopConstraintValue)
+            NSLayoutConstraint.activate([
+                self.leadingAnchor.constraint(equalTo: self.superview!.leadingAnchor, constant: 16),
+                self.topConstraint!
+            ])
+        } else {
+            self.topConstraint?.constant = calculatedTopConstraintValue
+        }
+    }
+
+    func parentEditingDidBegin() {
+        UIView.animate(withDuration: 0.1) {
+            self.textColor = self.activeColor
+            self.font = self.font.withSize(self.smallerFontSize)
+            self.topConstraint?.constant = self.intrinsicContentSize.height / -2
+            self.superview?.layoutIfNeeded()
+        }
+    }
+
+    func parentEditingDidEnd() {
```

Comment:
Make this more generic
- Rename to `moveToDefaultPosition()`
- This function should not check if the parent input is empty, let's move that responsibility to the parent. E.g. the `FloatingLabelTextField.editingEnded()` would look like the following
```
@objc func editingEnded() {
    borderLayer.changeBorderColor(active: false)
    if self.isInputEmpty() {
        self.floatingPlaceholderLabel.moveToDefaultPosition()
    } else {
        self.floatingPlaceholderLabel.updateTextColor(shouldHighlight: false)
    }
}
```
- As hinted from the above code snippet, this `moveToDefaultPosition` function should not change the text color if the input is empty, that also should be done from the parent.

Using generic function names and properties makes the code easier to read here and in call sites. Looking at the code snippet above, one can more easily tell what happens when the textfield ends editing compared to when the code was just
```
@objc func editingEnded() {
    borderLayer.setColor(isParentEditing: self.isEditing)
    self.floatingPlaceholderLabel.parentEditingDidEnd()
}
```